test/testkit/mocks/urlprovider/urlprovider.go

```diff
@@ -41,11 +42,11 @@ func (p *URLProvider) MockBackendExport(backendName string) string {
 	return p.mockBackendURL[backendName]
 }
 
-func (p *URLProvider) SetMetricPodURL(url string) *URLProvider {
-	p.metricPod = url
+func (p *URLProvider) SetMetricPodURL(podName, url string) *URLProvider {
+	p.metricPod[podName] = url
 	return p
 }
 
-func (p *URLProvider) MetricPodURL() string {
+func (p *URLProvider) MetricPodURL() map[string]string {
```

Comment:
The function name does not make sense since you return a map (many things). But maybe we can ditch that URL provider completely for the tests you touched in this PR?

Reply:
removed